#!/usr/bin/env python
PACKAGE = "pot_ground_segmentation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("hue_L",    int_t,    0, "Hue Lower threshold", 17,  0, 255)
gen.add("hue_H",    int_t,    0, "Hue Upper threshold", 100,  0, 255)

gen.add("sat_L",    int_t,    0, "Saturation Lower threshold", 10,  0, 255)
gen.add("sat_H",    int_t,    0, "Saturation Upper threshold", 100,  0, 255)

gen.add("val_L",    int_t,    0, "Value Lower threshold", 0,  0, 255)
gen.add("val_H",    int_t,    0, "Value Upper threshold", 65,  0, 255)


gen.add("contour_size",    int_t,    0, "Minimum contour size ", 100,  0, 1000)
gen.add("kernel_size",    int_t,    0, "Morphological opening kernel size", 5,  0, 11)



exit(gen.generate(PACKAGE, "pot_ground_segmentation", "HsvMask"))